lib/fun_with_flags/store/persistent/in_memory.ex

```diff
@@ -0,0 +1,163 @@
+defmodule FunWithFlags.Store.Persistent.InMemory do
+  alias FunWithFlags.{Config, Gate, Flag, Store.Persistent.Ecto.Record}
+
+  def worker_spec do
+    import Supervisor.Spec, only: [worker: 3]
+
+    worker(__MODULE__, [], restart: :permanent)
+  end
+
+  def child_spec(opts \\ []) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, [opts]}
+    }
+  end
+
+  def start_link(opts \\ []) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def init(_) do
+    {:ok, []}
+  end
+
+  def get(flag_name) do
+    GenServer.call(__MODULE__, {:get, flag_name})
+  end
+
+  def put(flag_name, gate) do
+    GenServer.call(__MODULE__, {:put, flag_name, gate})
+  end
+
+  def delete(flag_name, %Gate{} = gate) do
+    GenServer.call(__MODULE__, {:delete, flag_name, gate})
+  end
+
+  def delete(flag_name) do
+    GenServer.call(__MODULE__, {:delete, flag_name})
+  end
+
+  def all_flags do
+    GenServer.call(__MODULE__, :all_flags)
+  end
+
+  def all_flag_names do
+    GenServer.call(__MODULE__, :all_flag_names)
+  end
+
+  def handle_call({:get, flag_name}, _from, gates) do
+    {:reply, do_get(flag_name, gates), gates}
+  end
+
+  def handle_call({:put, flag_name, gate = %Gate{type: type}}, _from, gates) do
+    record = %Record{
```

Comment:
I don't think you need this. The [behaviour](https://github.com/tompave/fun_with_flags/blob/v1.4.0/lib/fun_with_flags/store/persistent.ex#L38-L39) says that these function only need to return `FunWithFlags.Flag{}` structs. You can store them directly in the GenServer with no need to transform them back and forth into `Record` structs.

Reply:
if I recall correctly, this was done because I needed to use the Ecto Serializer, which is only implemented for `Ecto.Record`. Not doing so would cause inconsistencies between tests and production (atoms vs strings, etc)